File: WeekOne/DayFour/inheritance.py
# inheritance.py

class Adult:
    def __init__(self, name, age, working, height, gender):
        self.name = name
        self.age = age
        self.working = working
        self.height = height
        self.gender = gender

    def work(self):
        print(f"I am going to work. My name is {self.name}.")

bob = Adult("Bob", 57, True, 185, "Male")
bob.work()

class Child(Adult):
    def __init__(self, name, age, working, height, gender, bottle):
        # super keyword allows the Child class to inherit all attributes and methods from the Adult class
        super().__init__(name, age, working, height, gender)
        self.bottle = bottle

    def use_bottle(self):
        if(self.bottle == True):
            print("Using bottle")
        else:
            print("You don't have a bottle.")

jimbob = Child("Jimbob", 16, True, 170, "Male", False) # bottle is unique to Child

jimbob.use_bottle()

# print(jimbob.bottle)
# print(bob.bottle) # AttributeError: Adult has no attribute 'bottle'

# jimbob.work()

File: WeekTwo/DayTwo/inheritance.js
class Adult {
    constructor(name, age, working, height, gender) {
        this.name = name
        this.age = age
        this.working = working
        this.height = height
        this.gender = gender
    }

    work() {
        console.log(`I am working. My name is ${this.name}`)
    }

}

const bob = new Adult("Bob", 57, true, 185, "Male")
// bob will ALWAYS be an instance of the Adult class
// the values can change but Bob the instance will always remain that
// bob = "cheese"
console.log(bob)
// bob.height = 186
// console.log(bob.height)
// bob.work()


class Child extends Adult {
    constructor(name, age, working, height, gender, bottle) {
        super(name, age, working, height, gender)
        this.bottle = bottle
    }

    use_bottle() {
        if (this.bottle == true) {
            console.log("Using bottle")
        }
        else {
            console.log("You don't have a bottle.")
        }
    }
}

// class Baby extends Child {
//     constructor(name, age, working, height, gender, bottle, burb) {
//         super(name, age, working, height, gender, bottle)
//         this.burb = burb
//     }
// }

// const tim = new Baby()

const jimbob = new Child("Jimbob", 16, true, 170, "Male", false)


//TODO: VEHICLE class with your own variables and 2 methods (functions)
//TODO: two inherited classes, that could be car and truck or car and plane with thir own methods too